docs/common-problems.asciidoc

```diff
@@ -0,0 +1,222 @@
+Open the **Logs** tab and select the `elastic_agent.apm_server` dataset.
+Look for any APM Server errors that could help diagnose the problem.
+
+**Can the {apm-agent} connect to {apm-server}**
+
+To determine if the APM agent can connect to the APM Server, send requests to the instrumented service and look for lines
+containing `[request]` in the APM Server logs.
+
+If no requests are logged, confirm that:
+
+. SSL isn't <<ssl-client-fails, misconfigured>>.
+. The host is correct. For example, if you're using Docker, ensure a bind to the right interface (for example, set
+`apm-server.host = 0.0.0.0:8200` to match any IP) and set the `SERVER_URL` setting in the APM agent accordingly.
+
+If you see requests coming through the APM Server but they are not accepted (a response code other than `202`),
+see <<common-response-codes>> to narrow down the possible causes.
+
+**Instrumentation gaps**
+
+APM agents provide auto-instrumentation for many popular frameworks and libraries.
+If the APM agent is not auto-instrumenting something that you were expecting, data won't be sent to the Elastic Stack.
+Reference the relevant {apm-agents-ref}/index.html[APM agent documentation] for details on what is automatically instrumented.
+
+[float]
+[[common-response-codes]]
+=== APM Server response codes
+
+[[bad-request]]
+[float]
+==== HTTP 400: Data decoding error / Data validation error
+
+The most likely cause for this error is using incompatible versions of APM agent and APM Server.
+See the <<agent-server-compatibility,agent/server compatibility matrix>> to verify compatibility.
+
+[[event-too-large]]
+[float]
+==== HTTP 400: Event too large
+
+APM agents communicate with the APM server by sending events in an HTTP request. Each event is sent as its own line in the HTTP request body. If events are too large, you should consider increasing the <<apm-input-general-settings,maximum size per event>>
+setting in the APM integration, and adjusting relevant settings in the agent.
+
+[[unauthorized]]
+[float]
+==== HTTP 401: Invalid token
+
+The <<secret-token,secret token>> in the request header doesn't match the secret token configured in the APM integration.
+
+[[forbidden]]
+[float]
+==== HTTP 403: Forbidden request
+
+Either you are sending requests to a <<apm-rum,RUM>> endpoint without RUM enabled, or a request
+is coming from an origin not specified in the APM integration settings.
+See the <<apm-input-rum-settings,Allowed origins>> setting for more information.
+
+[[request-timed-out]]
+[float]
+==== HTTP 503: Request timed out waiting to be processed
+
+This happens when APM Server exceeds the maximum number of requests that it can process concurrently.
+To alleviate this problem, you can try to: reduce the sample rate and/or reduce the collected stack trace information.
+See <<reduce-apm-storage>> for more information.
+
+Another option is to increase processing power.
+This can be done by either migrating your {agent} to a more powerful machine
```

Comment:
@elastic/apm-server This area doesn't mention that it may be because Elasticsearch is undersized or that the number of primary shards for the data streams is the default `1`. Should we mention these on this particular error?

It is something that our customers will see even if they increase processing power and add more APM Servers to the deployment, it may actually make things worse since the number of concurrent index requests will go up and only so many will be processed by Elasticsearch.